prefect_aws/s3.py

```diff
@@ -425,6 +425,15 @@ class S3Bucket(WritableFileSystem, WritableDeploymentStorage, ObjectStorageBlock
         ),
     )
 
+    cache_client: bool = Field(
+        default=False,
+        description=(
+            "If True, the S3 client will be cached. This is useful for "
+            "performance, but could cause issues if the S3 client is used "
+            "in multi-threaded environments."
+        ),
+    )
```

Comment:
I think this toggle makes more sense on the `AwsCredentials` class since it's closer to where the client is instantiated and allows other consumers of `AwsCredentials` to use the toggle. 

Also, do you know what issues client caching can cause in multi-threaded environments and if we can mitigate those issues? I'd prefer to not expose this toggle if we don't have to.